server/matchmaker/algorithm.py

```diff
@@ -52,13 +54,37 @@ def find(self) -> List[Match]:
                     search, preferred, search.quality_with(preferred),
                     search.match_threshold, preferred.match_threshold
                 )
-                if not search.matches_with(preferred):
-                    continue
 
                 self._propose(search, preferred)
 
+        self._forcefully_match_unmatched_newbies()
+
         return self._remove_duplicates()
 
+    def _forcefully_match_unmatched_newbies(self):
+        unmatched_newbies = [
+            search for search in self.searches 
+            if search.is_single_ladder_newbie()
+            and not search in self.matches
+        ] 
+
+        while 2 <= len(unmatched_newbies):
+            newbie1 = unmatched_newbies.pop()
+            newbie2 = unmatched_newbies.pop()
+            self._match(newbie1, newbie2)
+
+        if len(unmatched_newbies) == 1:
+            newbie = unmatched_newbies[0]
+            opponent = next(
+                (search for search in  self.searches
+                if search != newbie
+                and not search in self.matches),
+                None
```

Comment:
Woah. I didn't know `next` had a default parameter! Maybe passing it as a keyword argument could increase readability.

Reply:
I am not sure how to do that :( Trying to pass it as `next(iterator, default=None)` tells me that `TypeError: next() takes no keyword arguments`.

To make it clear that it's a default the following would work, but I'm not convinced it's clean...
 ```suggestion
           default_if_no_available_opponent = None
            opponent = next(
                (search for search in  self.searches
                if search != newbie
                and not search in self.matches),
                default_if_no_available_opponent
            )
            if opponent is not default_if_no_available_opponent:
                self._match(newbie, opponent)
```

Reply:
(An alternative is of course to try/except catch the `StopIterationError`, but that seemed less readable)

Reply:
Well it's not a big deal then.